### ISSUE TITLE
StaticClosureSniff: Closures using scope variables can not be made static

### ISSUE THREAD

**Issue by bendoh** [slevomat/coding-standard#738]
The following case demonstrates a situation where the `StaticClosureSniff` incorrectly marks a closure as static, even though doing so causes a fatal error:

```
<?php
/**
 * This file demonstrates the broken sniff in
 *    SlevomatCodingStandard\Sniffs\Functions\StaticClosureSniff
 */

declare(strict_types=1);

use PHPUnit\Framework\Assert;
use PHPUnit\Framework\TestCase;

class Handler
{
    public function handle($argument)
    {
        return $argument;
    }
}

class BrokenStaticClosureSniffTest extends TestCase
{
    public function test_sniff_breaks_this_test(): void
    {
        $handlerMock = $this->prophesize(Handler::class);
        $value = 'whee';

        // A fatal error is thrown if this is marked static
        $handlerMock->handle('foo')->will(function () use ($value) {
            return $value;
        });
        $handler = $handlerMock->reveal();

        Assert::assertEquals('whee', $handler->handle('foo'));
    }
}
```

The sniffer does this when asked to fix according to sniffs:

```
1) tests/unit/BrokenSniffTestCase.php

    ---------- begin diff ----------
--- Original
+++ New
@@ -25,7 +25,7 @@
         $value = 'whee';
 
         // A fatal error is thrown if this is marked static
-        $handlerMock->handle('foo')->will(function () use ($value) {
+        $handlerMock->handle('foo')->will(static function () use ($value) {
             return $value;
         });
         $handler = $handlerMock->reveal();

    ----------- end diff -----------

Applied checkers:

 * SlevomatCodingStandard\Sniffs\Functions\StaticClosureSniff.ClosureNotStatic
```

When then causes this error when you execute the test:

```
E                                                                   1 / 1 (100%)

Time: 117 ms, Memory: 8.00 MB

There was 1 error:

1) BrokenStaticClosureSniffTest::test_sniff_breaks_this_test
Cannot bind an instance to a static closure
```

I believe this means that scope variables for instance methods that are imported into the closure scope via `use` render the closure unable to be marked static, and checking for references to `$this` or `parent::` is insufficient to determine if the closure can be marked as static, but it also needs to check for a `use` keyword in the closure definition.

**Comment by kukulich** [comment]
It's not fixable.
